src/Network/NWPathMonitor.cs

```diff
@@ -91,6 +105,12 @@ public void SetUpdatedSnapshotHandler (Action<NWPath> callback)
 				}
 			}
 		}
+		Action<NWPath> userSnapshotHandler;
+
+		public void SetUpdatedSnapshotHandler (Action<NWPath> callback)
```

Comment:
I would Obsolete the method and add a Property as the new API.
```suggestion
		Action<NWPath> SnapshotHandler { 
			get => userSnapshotHandler; 
			set => userSnapshotHandler = value; 
		}
		
		[Obsolete ("Use the 'SnapshotHandler' property instead.")]
		public void SetUpdatedSnapshotHandler (Action<NWPath> callback)
```

Reply:
my code is missing ```public```